### ISSUE TITLE
ARIA Grid: Applications Mode will not auto activate for an ARIA Grid

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3743]
**Reported by bgaraventa on 2014-01-02 23:59**
When an ARIA Grid construct uses all requisite ARIA roles and states to convey an ARIA Grid, such as aria-owns and proper role nesting, it is not possible to use Tab to navigate into an ARIA Grid and maintain Applications Mode so that the arrow keys can be used to navigate the Grid.

NVDA automatically disables Applications Mode when the Grid receives focus, and the programmed keyboard functionality cannot be used without forcibly disabling Virtual Buffer Mode using Insert+Space, which is non-intuitive.

This occurs in the latest version of Firefox.

Live demo:
http://whatsock.com/test/demo/Coding%20Arena/ARIA%20Data%20Grids/ARIA%20Data%20Grid%20(Dynamic)/demo.htm

Steps to reproduce:
1. Navigate to the test page.
2. Use Tab to set focus into the Grid.
3. Confirm that Applications Mode is not activated, and the arrow keys cannot be used to navigate the Grid.

The above ARIA Grid construct is programmed according to the spec at
http://www.w3.org/TR/wai-aria/usage#managingfocus
And follows the guidelines documented at
https://www.ssbbartgroup.com/blog/2013/10/22/why-there-are-only-two-ways-to-make-aria-widgets-programmatically-focusable-for-screen-reader-users/

Also, aria-readonly="true" shouldn't automatically disable Applications Mode if this is a determining factor, since this will cause the Grid to lose focus when a readonly row header cell is encountered for instance.
( http://www.w3.org/TR/2011/CR-wai-aria-20110118/states_and_properties#aria-readonly )

**Comment by leonardder** [comment]
@MichaelDCurran, Since you've been working with Aria Grids recently (#7410), you probably must have seen the behaviour in this ticket when it still occurs, and whether it is unexpected.

**Comment by jcsteh** [comment]
Duplicate of #5118, fixed in a50699cf.
